cmd/cadvisor.go

```diff
@@ -124,6 +149,31 @@ func main() {
 	klog.V(1).Infof("enabled metrics: %s", includedMetrics.String())
 	setMaxProcs()
 
+	plugins := map[string]container.Plugin{
+		"containerd": containerd.NewPluginWithOptions(&containerd.Options{ // TODO(@spartan0x117): Make containerdEnvMetadataWhiteList non-global and add to these options
+			ContainerdEndpoint:  *ArgContainerdEndpoint,
+			ContainerdNamespace: *ArgContainerdNamespace,
+		}),
+		"docker": docker.NewPluginWithOptions(&docker.Options{
+			DockerEndpoint: *ArgDockerEndpoint,
+			DockerTLS:      *ArgDockerTLS,
+			DockerCert:     *ArgDockerCert,
+			DockerKey:      *ArgDockerKey,
+			DockerCA:       *ArgDockerCA,
+		}),
+		"crio": crio.NewPlugin(), // TODO(@spartan0x117): Add options
```

Comment:
Looked through the crio code, the [values](https://github.com/google/cadvisor/blob/master/container/crio/factory.go#L33) are currently `const`s, so making these configurable options would be a departure from the current behavior. Happy to do it, but my thinking is that we want to keep the PR smaller.
```suggestion
		"crio": crio.NewPlugin(),
```

Reply:
Yeah, they look like either const or unexported. Since we also don't need any of them on the Agent side, I think it's OK leaving them as-is for now.